shell/platform/android/io/flutter/embedding/engine/android/AndroidTouchProcessor.java

```diff
@@ -0,0 +1,186 @@
+package io.flutter.embedding.engine.android;
+
+import android.support.annotation.NonNull;
+import android.view.MotionEvent;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+import io.flutter.embedding.engine.renderer.FlutterRenderer;
+
+/**
+ * Sends touch information from Android to Flutter in a format that Flutter
+ * understands.
```

Comment:
Suggestion: describe how you're transforming touch events to fit them into Flutter.

Especially: describe caveats, or places where information gets lost in translation.